RuntimeUnityEditor/Windows/ObjectTree/ObjectTreeViewer.cs

```diff
@@ -102,27 +103,34 @@ private void DisplayObjectTreeHelper(GameObject go, int indent, ref int currentC
         {
             currentCount++;
 
-            var needsHeightMeasure = _singleObjectTreeItemHeight == 0;
+            var needsHeightMeasure = _singleObjectTreeItemHeight == 0f;
 
-            var isVisible = currentCount * _singleObjectTreeItemHeight >= _treeScrollPosition.y &&
-                            (currentCount - 1) * _singleObjectTreeItemHeight <= _treeScrollPosition.y + _objectTreeHeight;
+            var isVisible = currentCount * _singleObjectTreeItemHeight + _singleObjectTreeItemMargin >= _treeScrollPosition.y &&
+                            (currentCount - 1) * _singleObjectTreeItemHeight + _singleObjectTreeItemMargin <= _treeScrollPosition.y + _objectTreeHeight;
 
-            if (isVisible || needsHeightMeasure || _scrollTreeToSelected)
+            if (SelectedTransform == go.transform && _scrollTreeToSelected)
+            {
+                _scrollTarget = currentCount == 1 ? 0 : _singleObjectTreeItemHeight * (currentCount - 1);
+            }
+
+            if (isVisible || needsHeightMeasure)
```

Comment:
Nice performance, never need to draw the entire tree by removing _scrollTreeToSelected and moving above